### DISCUSSION TITLE
Dealing with cyclic dependencies

### DISCUSSION THREAD

**Discussion by ghost** [pydantic/pydantic#2395] [Feature Request]
We need a way to deal with cyclic dependencies. My suggestion is to provide a convenient way to import modules late.

```python

def do_import():
    from machine import MachineState # machine module may import this one

class MachineReport(BaseModel):
    class Config:
        late_import = do_import

    state: 'MachineState'
   
```

**Comment by ssbarnea** [comment]
I had to do this yesterday, is [documented](https://pydantic-docs.helpmanual.io/usage/postponed_annotations/#self-referencing-models). Take a look at:

https://github.com/ansible-community/schemas/blob/main/src/ansibleschemas/tasks.py#L83-L84

**Reply by timorieber** [reply]
It seems this only works for models defined within the same module, but not from another package. What you're describing isn't a cyclic import, but a forward reference to a model which is implemented later in the same code.

**Comment by silllli** [comment]
This seems to continue the discussion from #1873, right?
